### ISSUE TITLE
keras model evalueate calculated loss error

### ISSUE THREAD

**Issue by chaoyan1037** [keras-team/keras#8952]
I found a strange problem. When I use as keras.losses.binary_crossentropy as the model loss, the loss I get from the model.evaluate API is very different from that calculated using the keras.losses.binary_crossentropy API.

For example:
```
    # use keras.losses.binary_crossentropy as loss
    opt = Adam(lr=learning_rate, beta_1=0.9, beta_2=0.999, epsilon=1e-08)
    model.compile(loss=keras.losses.binary_crossentropy, optimizer=opt, metrics=['accuracy'])
    ...
    # get loss and test accuracy from evaluate API
    test_logloss, test_acc = model.evaluate(test_img, test_label, verbose=0, batch_size=batch_size)
    
  
    test_predict = model.predict(test_img, verbose=0, batch_size=batch_size)
    test_predict_tf = tf.convert_to_tensor(test_predict)
    # get loss and test accuracy from binary_crossentropy API
    test_loss2 = keras.losses.binary_crossentropy(test_label, test_predict_tf)
    test_loss2 = tf.Session().run(test_loss2)
```
Does any one know why?

**Comment by fredxfred** [comment]
I am getting the same problem. Even worse, when I initialize a model to simplify perform the identity operation on the input, my model.evaluation() is nonzero, yet my predicted values are exactly equivalent! This has to be a bug. I wrote this script to make sure I wasn't going insane:
```
from keras.layers import Input, Dense
from keras.models import Model
from keras.models import Sequential
import numpy as np

n = 2
dummy_data = np.random.random((10, n)).astype(np.float32)

x = Sequential()
x.add(Dense(n, kernel_initializer='identity', activation='linear', input_dim=n))
x.compile(optimizer='adadelta', loss='binary_crossentropy')

y = Sequential()
y.add(Dense(n, kernel_initializer='identity', activation='linear', input_dim=n))
y.compile(optimizer='adadelta', loss='mean_absolute_error')

x_pred = x.predict(dummy_data)
y_pred = y.predict(dummy_data)
for i in range(10):
    print(dummy_data[i], x_pred[i], y_pred[i])
print("x eval:", x.evaluate(dummy_data, dummy_data))
print("y eval:", y.evaluate(dummy_data, dummy_data))
print("Types:", dummy_data.dtype, x_pred.dtype, y_pred.dtype)
```
Which has output:
```
[0.36412117 0.22953235] [0.36412117 0.22953235] [0.36412117 0.22953235]
[0.5420893  0.24696268] [0.5420893  0.24696268] [0.5420893  0.24696268]
[0.40555134 0.8496417 ] [0.40555134 0.8496417 ] [0.40555134 0.8496417 ]
[0.9258905 0.7616086] [0.9258905 0.7616086] [0.9258905 0.7616086]
[0.74589723 0.20100497] [0.74589723 0.20100497] [0.74589723 0.20100497]
[0.14783198 0.22654009] [0.14783198 0.22654009] [0.14783198 0.22654009]
[0.1369148 0.6688388] [0.1369148 0.6688388] [0.1369148 0.6688388]
[0.56828284 0.49896967] [0.56828284 0.49896967] [0.56828284 0.49896967]
[0.06548529 0.13393487] [0.06548529 0.13393487] [0.06548529 0.13393487]
[0.32614648 0.7063883 ] [0.32614648 0.7063883 ] [0.32614648 0.7063883 ]
10/10 [==============================] - 0s 646us/step
x eval: 0.5330568552017212
10/10 [==============================] - 0s 492us/step
y eval: 0.0
Types: float32 float32 float32
```

This is with a Tensorflow backend. I made sure to update everything to the latest version. I also found that if I attempt to fit the above models x and y using the dummy_data as both training data and labels, that x will change its values while y will stay the same (as it should - there is no training error). My guess is that the bug has something to do with mismatched datatypes.
